sim/src/kernels/ocean_noise.opencl

```diff
@@ -0,0 +1,292 @@
+	float2 u = ansic_lcg(&seed);
+
+	seed_row[ti] = seed;
+
+	if (2*ti + 1 < random_nx_) {
+	    random_row[2*ti    ] = u.x;
+	    random_row[2*ti + 1] = u.y;
+	}
+	else if (2*ti == random_nx_) {
+	    random_row[2*ti    ] = u.x;
+	}
+    }
+}
+
+
+/**
+  * Kernel that generates normal distributed random numbers with mean 0 and variance 1.
+  */
+__kernel void normalDistribution(
+        // Size of data
+        int seed_nx_, int seed_ny_,
+	int random_nx_, 
+	
+        //Data
+        __global float* seed_ptr_, int seed_pitch_,
+        __global float* random_ptr_, int random_pitch_
+    ) {
+
+    //Index of cell within domain
+    const int ti = get_global_id(0); 
+    const int tj = get_global_id(1);
+
+    // Each thread computes and writes two uniform numbers.
+
+    if ((ti < seed_nx_) && (tj < seed_ny_)) {
+    
+	//Compute pointer to current row in the U array
+	__global float* const seed_row = (__global float*) ((__global char*) seed_ptr_ + seed_pitch_*tj);
+	__global float* const random_row = (__global float*) ((__global char*) random_ptr_ + random_pitch_*tj);
+	
+	float seed = seed_row[ti];
+	float2 u = boxMuller(&seed);
+
+	seed_row[ti] = seed;
+
+	if (2*ti + 1 < random_nx_) {
+	    random_row[2*ti    ] = u.x;
+	    random_row[2*ti + 1] = u.y;
+	}
+	else if (2*ti == random_nx_) {
+	    random_row[2*ti    ] = u.x;
+	}
+    }
+}
+
+
+/**
+  * Local function calculating the SOAR function given two grid locations
+  */
+float soar_covariance(int a_x, int a_y, int b_x, int b_y,
+		      float dx, float dy, float soar_q0, float soar_L) {
+    const float dist = sqrt( dx*dx*(a_x - b_x)*(a_x - b_x) +
+			     dy*dy*(a_y - b_y)*(a_y - b_y) );
+    return soar_q0*( 1.0f + dist/soar_L)*exp(-dist/soar_L);
+}
+
+
+
+/**
+  * Kernel that adds a perturbation to the input field eta.
+  * The perturbation is based on a SOAR covariance function using a cut-off value of 2.
+  */
+__kernel void perturbOcean(
+        // Size of data
+	int nx_, int ny_,
+        float dx_, float dy_,
+	int ghost_cells_x_, int ghost_cells_y_,
+	
+	// physical parameters
+	float g_, float f_, float beta_, float y0_reference_cell_,
+	
+	// Parameter for the SOAR function
+	float soar_q0_, float soar_L_, 
+	
+	// Periodic domain
+	int periodic_north_south_, int periodic_east_west_,
+	
+        // random data
+	__global float* random_ptr_, int random_pitch_,
+
+	// Ocean data
+	__global float* eta_ptr_, int eta_pitch_,
+	__global float* hu_ptr_, int hu_pitch_,
+	__global float* hv_ptr_, int hv_pitch_,
+	__global float* Hi_ptr_, int Hi_pitch_
+    ) {
+
+    //Index of cell within block
+    const int tx = get_local_id(0); 
+    const int ty = get_local_id(1);
+
+    //Index of start of block within domain
+    const int bx = get_local_size(0) * get_group_id(0);
+    const int by = get_local_size(1) * get_group_id(1);
+
+    //Index of cell within domain
+    const int ti = get_global_id(0);
+    const int tj = get_global_id(1);
+
+    const int cutoff = 2;
```

Comment:
Should be an argument to the kernel (even though we most likely will not change it).